### PR TITLE
Feature | WalletChat.fun Widget Installation

### PR DESCRIPTION
**Pull request by cryptoKevinL**
# Description
WalletChat Widget will point to gooddollar.walletchat.fun, and new users signing up on gooddapp.org will enter a GoodDollar general chat room (

If the GoodDollar Team wishes to have a support wallet to handle support requests within the WalletChat Widget, we can accommodate this.

New dependency added for NPM package react-wallet-chat-gooddollar

About # (link your issue here)

# How Has This Been Tested?
We have tested this locally, in addition we have (https://good-protocol-ui.vercel.app/#/swap) live which can be used to test the changes proposed.

Please describe the tests that you ran to verify your changes.
We tested with multiple developers to install, send messages and see that the widget works as expected with the new site create for gooddollar.

image

We have a Telegram group setup with Hadar @sirpy - we look forward to the integration and are happy to help with any questions.

### CODE REVIEW

**Review by sirpy — CHANGES_REQUESTED** [review]
@cryptoKevinL 
1. this needs to work with our wallet provider so user doenst need to double connect his wallet
2. needs to function properly in mobile view
3. need some kind of backoffice for support team to handle access

**Comment by johnsmith-gooddollar** [comment]
@sirpy @cryptoKevinL  which is status of this ? it's not been updated since March

File: src/pages/App.tsx [hunk 1]
```diff
@@ -61,13 +65,60 @@ function App(): JSX.Element {
...
     }, [pathname, sendData])
 
+    // useEffect(() => {
+    //     //Test code for adding signature after wallet connection
+    //     if (wallet) {
+    //         const ethersProvider = new ethers.providers.Web3Provider(wallet.provider, 'any')
+
+    //         const signer = ethersProvider.getSigner()
+    //         signer
+    //             .getAddress()
```
Comment (johnsmith-gooddollar) [inline src/pages/App.tsx:89]: if this not needed anymore let's remove it

**Review by johnsmith-gooddollar — APPROVED** [review]

File: src/pages/App.tsx (hunk 1 above)
Comment (cryptoKevinL) [reply src/pages/App.tsx]: comments have been cleaned up, thanks!  Are there any other issues needed to fix before deployment?

File: src/pages/App.tsx [hunk 2]
```diff
@@ -16,6 +16,8 @@ import { useFaucet } from '@gooddollar/web3sdk-v2'
 import TransactionUpdater from '../state/transactions/updater'
```
Comment (johnsmith-gooddollar) [inline src/pages/App.tsx:16]: I see a lot of stuff been added to App, let's move it to separate component `src/components/walletchat/WalletChat.js`:

```javascript
import React from 'react'

import { WalletChatWidget } from 'react-wallet-chat'
import { useConnectWallet, useSetChain } from '@web3-onboard/react'
import { isMobile } from '../../lib/utils/platform'

const WalletChat = () => {
  const [{ chains, connectedChain }] = useSetChain()
   const [{ wallet }] = useConnectWallet()
   const connectedAccount = wallet?.accounts[0]
   const activeChain = chains.find((chain) => chain.id === connectedChain?.id)
   const chainId = activeChain?.token === 'CELO' ? 42_220 : 1

  return (
    <WalletChatWidget
         style={{ marginBottom: isMobile ? '75px' : '0px' }}
         connectedWallet={
             connectedAccount &&
             chainId && {
                 walletName: wallet.label,
                 account: connectedAccount.address,
                 chainId,
             }
         }
     />
  )
}

export default WalletChat

Comment (johnsmith-gooddollar) [reply src/pages/App.tsx:16]: also not sure you need for onboard and connect wallet. we have own wallet could be accessible through useWallet() hook

import { useWallet } from 'src/lib/wallet/GoodWalletProvider'

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: src/pages/App.tsx (hunk 2 above)
Comment (johnsmith-gooddollar) [reply src/pages/App.tsx:16]: useActiveWeb3React will return you 

```typescript
chainId: ChainId
label?: string
```

also it returns account - see AppBar.tsx
so it's possible to use this hook instead

Comment (cryptoKevinL) [reply src/pages/App.tsx:16]: @johnsmith-gooddollar - This should be resolved, good suggestions!

**Comment by cryptoKevinL** [comment]
Also @sirpy or @johnsmith-gooddollar please provide me a wallet address which your support team could monitor (Direct Support DMs to this address within WalletChat), and I can update this.  Its just an ENV on our side so not required to be part of this PR.

**Comment by sirpy** [comment]
@patpedrosa please review feature with Lewis or @cryptoKevinL

**Review by johnsmith-gooddollar — APPROVED** [review]

**Comment by cryptoKevinL** [comment]
@sirpy with your permission, I think we are ready to test this out.  We'd like to see what usage/feedback it gets.

**Comment by L03TJ3** [comment]
https://github.com/GoodDollar/GoodProtocolUI/pull/439

Got merged through separate PR to pull and sync with Vite changes

**Pull request closed without merging**